Sources/GDBRemote/SessionBase.cpp

```diff
@@ -158,12 +158,12 @@ bool SessionBase::sendError(ErrorCode code) {
     break;
   }
 
-  char message[4];
-  message[0] = 'E';
-  message[1] = NibbleToHex(code >> 4);
-  message[2] = NibbleToHex(code & 15);
-  message[3] = '\0';
-  return send(message);
+  std::ostringstream ss;
+  ss << "E";
+  addPacketSeparator(ss);
+  ss << std::hex << code;
```

Comment:
This is not good, you're changing the packet format. `std::hex` doesn't add `0` at the beginning to always have a two character-long error code.